src/spdx/parser/rdf/graph_parsing_functions.py

```diff
@@ -102,3 +102,44 @@ def remove_prefix(string: str, prefix: str) -> str:
     if string.startswith(prefix):
         return string[len(prefix) :]
     return string
+
+
+def get_correctly_typed_triples(
+    logger: Logger,
+    graph: Graph,
+    subject: Optional[Node] = None,
+    predicate: Optional[Node] = None,
+    _object: Optional[Node] = None,
+) -> Tuple[URIRef, Node, Union[BNode, Literal, URIRef]]:
+    # this is a helper method to enforce some rdf types for the triples to bo compatible with the
+    # code that follows
+    for s, p, o in graph.triples((subject, predicate, _object)):
+        if not isinstance(s, (BNode, URIRef)):
```

Comment:
We allow `BNode` here but the type hint for the subject is only `URIRef`

Reply:
Good catch